umake/frameworks/__init__.py

```diff
@@ -331,6 +334,14 @@ def run_for(self, args):
                        auto_accept_license=auto_accept_license,
                        dry_run=dry_run)
 
+    def get_latest_version(self):
+        return (re.search(self.version_regex, self.package_url).group(1).replace('_', '.')
+            if self.package_url and self.version_regex else None)
+
+    @staticmethod
+    def get_current_user_version(install_path):
+        return None
+
```

Comment:
Adding two default methods for each framework.
get_latest_version() is parsing the download url string to extract the version.
get_current_user_version() is overridden in each of the framework. It is checking the installed files to fetch the version.